sdk/python/kubeflow/training/api/training_client.py

```diff
@@ -81,6 +89,141 @@ def __init__(
             )
         self.job_kind = job_kind
 
+    def train(
+        self,
+        name: str = None,
+        namespace: str = None,
+        num_workers: int = 1,
+        num_procs_per_worker: int = 1,
+        storage_config: Dict[Literal["size", "storage_class"], str] = None,
+        model_provider_parameters: HuggingFaceModelParams = None,
+        dataset_provider_parameters: S3DatasetParams = None,
+        train_parameters: HuggingFaceTrainParams = None,
+        resources_per_worker: Dict[Literal["gpu", "cpu", "memory"], any] = None,
+    ):
+        """
+        Higher level train api
+        """
+        if (
+            not name
+            or not namespace
+            or not storage_config
+            or not model_provider_parameters
+            or not dataset_provider_parameters
+            or not train_parameters
+            or not resources_per_worker
+        ):
+            raise ValueError("One of the required parameters is None")
+
+        try:
+            self.core_api.create_namespace(
+                body=utils.get_namespace_spec(namespace=namespace)
+            )
+        except Exception as e:
+            print(e)
```

Comment:
Why do we need to create namespace as part of this API ?

Reply:
Since we don't want user to deal with Kubernetes side of things, namespace creation is also added so that job doesn't fail

Reply:
I think, we should avoid the namespace creation as part of this API. Otherwise, we need to give Kubeflow user permission to create namespaces which might cause confusions.

Usually, user namespaces should be pre-created before using the Training Operator SDK APIs since it might require additional orchestration (e.g. resource allocation, access setup, etc.)
For example, Kubeflow Profile Controller is responsible to create namespaces for Kubeflow users (cc @kimwnasptd).

Reply:
Okay